mtp_bank_admin/apps/bank_admin/templatetags/dates.py

```diff
@@ -11,3 +12,9 @@ def get_preceding_days(number_of_days, offset=0):
     current = now().date()
     days = [current - timedelta(days=day) for day in range(offset, offset + number_of_days)]
     return days
+
+
+@register.assignment_tag
+def get_preceding_weekdays(number_of_days, offset=0):
+    days = get_preceding_days(int(math.ceil(number_of_days * 7 / 5)) + 1, offset=offset)
```

Comment:
As get_preceding_days isn't used as an actual template tag you could just make it a helper function which returns an unlimited generator, which the filter below would deal with taking the correct number of days from without needing to do all the arithmetic in this line. Might just be a bit prettier I think, but up to you.